### ISSUE TITLE
ONNX-ResNet50 CPP export

### ISSUE THREAD

**Issue by andreistoian** [CEA-LIST/N2D2#71]
I tried importing the ONNX ResNet50v2 model and exporting it to CPP:  n2d2 "models/ONNX/resnet-50-v2-onnx.ini" -test -seed 1 -export CPP -nbbits -32. I disabled the database since I don't have ILSVRC2012. As SoftMax is not supported for export I changed the ini file to set the output to:

[resnetv24_dense0_fwd.Target]
TopN=5

I get an _bad_function_call error for a cell that is N2D2::BatchNormCell_frame_cuda at 
std::unique_ptr<N2D2::CPP_CellExport> N2D2::CPP_CellExport::getInstance(Cell& cell)
{
    return Registrar<CPP_CellExport>::create(cell.getType())(cell);
}

The log tail is : 
-> Generating cell resnetv24_stage4__plus2
-> Generating cell resnetv24_batchnorm2_fwd
-> Generating cell resnetv24_pool1_fwd
-> Generating cell resnetv24_dense0_fwd
Invalid registrar key "BatchNorm"

**Comment by davidbriand-cea** [comment]
Hi,
Thank you for the issue !

The pure CPP export doesn't support BatchNormalization layers. Only layers that can be fully-quantized as integers operators (even in 32-bits floating point export mode) are supported under the CPP model export. This is why softmax is not supported in the CPP export. 
If you want to export ResNet-50 model under a CPP export i recommand you to use the  "resnet-50-v1-onnx.ini". In this model the BatchNormalization parameters will be automatically fused with weights and biases of their associated convolution layers by N2D2 before export generation. Fusion of BatchNormalization parameters with convolution is not possible with the resnet-50-v2 model because batchnormalization is the first layer of the network and also because each element wise is followed with batchnormalization that cannot be fused.  

Hope this answer can help you

Regards

**Comment by davidbriand-cea** [comment]
Also don't forget to add "-w /dev/null" in your export command :  
n2d2 "models/ONNX/resnet-50-v1-onnx.ini" -test -seed 1 -export CPP -nbbits -32 -w /dev/null
This will load weights stores in the ONNX model.

Regards
